### PR TITLE
Issue 2911

### PR DESCRIPTION
**Pull request by akumar-cs**
pr against master to compare changes with containerized templates ,once design is validated will. marking pr as draft for that purpose

### CODE REVIEW

File: models/h2o_dai_regression_auto_insurance/app_h2o_mojo_pipeline_gunicorn.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        """
+        overridden method to make external global imports
+        When using external imports override the method to provide necessary imports. Make sure to mark them `global` to
+        be used by certifai interpreter correctly. set once,use throughout.
+        :return: None
+        """
+        global dt
+        import datatable as dt
+
+    def _get_prediction_class(self, preds):
```
Comment (cdraper-CS) [inline models/h2o_dai_regression_auto_insurance/app_h2o_mojo_pipeline_gunicorn.py:53]: I think we should rename the function to be _get_prediction given what it does in the regression case is return a probability not the class

File: models/containerized_model/templates/prediction_service_h2o_mojo.py [hunk 2]
```diff
@@ -39,27 +30,51 @@ def read_and_save_file(source_path, destination_path):
...
+        overridden method to make external global imports
+        When using external imports override the method to provide necessary imports. Make sure to mark them `global` to
+        be used by certifai interpreter correctly. set once,use throughout.
+        :return: None
+        """
+        global dt
+        import datatable as dt
+
+    # <Fill in this function and remove the 'raise'>
+    def _get_prediction_class(self, preds):
```
Comment (cdraper-CS) [inline models/containerized_model/templates/prediction_service_h2o_mojo.py:62]: It's highly desirable to have a "will usually work for H2O" implementation here, and have the user override it only if needed using a list of the outcome classes in the order they will be returned (c.f. columns).   We can then move the overrides into metadata that's optionally provided with the model but that'll need more refactoring of containerization scripts so perhaps leave that out of this PR.

Also think we should rename the function to be _get_prediction given what it does in the regression case is return a probability not the class

**Review by cdraper-CS — COMMENTED** [review]
Do any of these changes require 1.3.6, or is that only when the user turns on production=True?

File: models/containerized_model/templates/prediction_service_h2o_mojo.py (hunk 2 above)
Comment (akumar-cs) [reply models/containerized_model/templates/prediction_service_h2o_mojo.py:62]: all these are 1.3.6. like gunicorn won't work with 1.3.5  if any of the models require external imports like xgb and datatable (in-case of h20). I can add that docs specifying these examples work with 1.3.5 and we should keep this in a separate branch for now and merge as and when required

Comment (akumar-cs) [reply models/containerized_model/templates/prediction_service_h2o_mojo.py:62]: _get_prediction for regression returns actual prediction values like a floating point  number not probabilities, so yea i should rename it to get_prediction

Comment (akumar-cs) [reply models/containerized_model/templates/prediction_service_h2o_mojo.py:62]: h2o that works out of box will require us to either fix deafult case as binary classification or accept metadata file

Comment (cdraper-CS) [reply models/containerized_model/templates/prediction_service_h2o_mojo.py:62]: @akumar-cs Why can't we figure it out from the preds (1, 2 or more results per...)?

**Comment by cdraper-CS** [comment]
Have incorporated these changes into other PRs and merged work for xgboost and H2O regression. Remaining open one is: https://github.com/CognitiveScale/cortex-certifai-examples/pull/67

**Comment by cdraper-CS** [comment]
Closng as superceded by other PR

**Pull request closed without merging**